### ISSUE TITLE
Pyright incorrectly reports incompatibleMethodOverride for property overriden with cached_property

### ISSUE THREAD

**Issue by gotmax23** [microsoft/pyright#10252]
**Describe the bug**

Pyright incorrectly reports incompatibleMethodOverride for property overriden with cached_property. This does not happen in mypy.

**Code or Screenshots**

Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBMCuB2BjALmMAbAzlAlhADmCMlIgIaIAWApgCYD6%2B4%2B1xAngFCiRTJv454Ac1wEiJAArhUiDBw6J0ZTNilgZGAMJKVACjUb0ASgBcHKBagABJmBbtzl2tWBRb%2BXZmrpgRqAFoAPihMZBATKAA6aPlnV3pMSGp6agAPFnhMHAA3ZIpkHDB4XT8gkLCzSygQamRYEHgoACIm%2BUVlbABJAnR9aTBZdG0O00cLK3IqOkZmVj4xqDi3Zk9vXwDg0PCFqpq6hqgEpJT06kycvJRC4qMgA)

```python
from functools import cached_property
from typing import Protocol

class ProtocolClass(Protocol):
    @property
    def prop(self) -> str: ...

def _some_expensive_action() -> str:
    return ""

class Impl(ProtocolClass):
    @cached_property
    def prop(self) -> str:
        return _some_expensive_action()
```

``` console
$ uvx pyright cached_property.py
/home/gotmax/dev/repros/cached_property.py
  /home/gotmax/dev/repros/cached_property.py:16:9 - error: "prop" incorrectly overrides property of same name in class "ProtocolClass" (reportIncompatibleMethodOverride)
1 error, 0 warnings, 0 informations 
```

**VS Code extension or command-line**

``` console
$ uvx pyright --version         
pyright 1.1.398
```

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

The type of a `property` is not the same as a `cached_property`. They have similar functionality, but from a type standpoint they have no overlap (i.e. there is no common base class between the two). 

Pyright has significant special casing for properties because the `property` class definition is not generic (for historical reason). It has no special casing for `functools.cached_property`, so it treats it like any other class.

If you want to use a `cached_property` to override a `property` in a parent class and you know this is safe for your use case, you can use a `# type: ignore` or `# pyright: ignore` to suppress the diagnostic.

**Comment by analog-cbarber** [comment]
Perhaps this is "as designed", but the behavior is not ideal for users, especially in the case where the superclass is abstract. Sure, you can always add a `# type: ignore`, but now you will be ignoring not only the difference between `@cached_property` and `@property` but also ignoring any difference in the declared return type.

Of course you can get around this by redirecting to another property in the subclass, e.g.:

```python
class Impl(ProtocolClass):
    @property
    def prop(self) -> str:
        return self._prop

    @cached_property
    def _prop(self) -> str:
        return _some_expensive_action()
```

Pyright is perfectly happy with this, but it isn't clear that this extra verbosity and extra function call makes your code better.

So even if this is really your intent, this seems like an annoying wart. 

FWIW, mypy chose to treat`@cached_property` as being compatible with `@property`.
